### PR TITLE
Update Release pipelines to includes Debian 12

### PR DESCRIPTION
**Pull request by yophilav**
Update Release pipeline with Debian 12: 
- Added artifacts building within Debian12 base image
- Publish ARMHF architecture to Debian12 PMC
  - https://packages.microsoft.com/debian/12/prod/
  - https://packages.microsoft.com/repos/microsoft-debian-bookworm-prod/ 

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

**Comment by yophilav** [comment]
Remark: The Release pipeline is shown as "Failed" here because I canceled the pipeline after it successfully build the packages. We don't want it to publish anything to the PMC yet.

File: builds/misc/packages-release.yaml [hunk 1]
```diff
@@ -202,6 +215,14 @@ stages:
           pmcRepoName: microsoft-debian-bullseye-prod-apt
           pmcRelease: bullseye
 
+        Debian12-arm32v7:
```
Comment (damonbarry) [inline builds/misc/packages-release.yaml:218]: We've decided to promote all three architectures to Tier 1, so they all need to be published to PMC (and GitHub).

**Comment by damonbarry** [comment]
Thanks for doing this! FYI, I have some changes in a branch that will add Debian 12 to the CI build, tests, etc., just was waiting to get the 1ES agent image in place first.

**Pull request closed without merging**